src/components/StarRating/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { ReactComponent as Star } from 'assets/svg/star.svg';
+import {
+  createContext, useMemo, useState,
+} from 'react';
+import useContext from '../../utils/hooks/useContext';
+import styles from './StarRating.module.scss';
+
+interface StarRateContextType {
+  enter: () => void;
+  leave: () => void;
+  click: (num: number) => void;
+  setActive?: () => void;
+}
+interface Props {
+  setActive: () => void;
+}
+
+export const StarRateContext = createContext<StarRateContextType | null>(null);
+
+function LeaveStarRateContainer({ rating }: { rating: number }) {
+  const { enter } = useContext(StarRateContext);
+  return (
+    <div
+      className={styles.starRateContainer}
+      onMouseEnter={enter}
+    >
+      { [1, 2, 3, 4, 5].map((num) => (
+        <button
+          key={num}
+          type="button"
+          className={styles.wrapper}
+        >
+          <Star
+            key={num}
+            className={styles.wrapper__star}
+            fill={rating >= num ? '#ff7f23' : '#eee'}
+          />
+        </button>
+      ))}
+    </div>
+  );
+}
+
+function EnterStarRateContainer({ rating }: { rating: number }) {
+  const {
+    leave, click, setActive,
+  } = useContext(StarRateContext);
+  const [starNum, setStarNum] = useState(rating);
+
+  return (
+    <div
+      className={styles.starRateContainer}
+      onMouseLeave={() => {
+        leave();
+        setStarNum(0);
+      }}
+    >
+      {[1, 2, 3, 4, 5].map((num) => (
+        <button
+          key={num}
+          type="button"
+          className={styles.wrapper}
+          onMouseEnter={() => setStarNum(num)}
+          onClick={() => {
+            click(num);
+            setActive?.();
+          }}
+        >
+          <Star
+            key={num}
+            className={styles.wrapper__star}
+            fill={starNum >= num ? '#ff7f23' : '#eee'}
+          />
+        </button>
+      ))}
+    </div>
+  );
+}
+
+export default function StarRating({ setActive }: Props) {
+  const [rating, setRating] = useState(0);
+  const [mouseEnter, setMouseEnter] = useState(false);
+
+  const value = useMemo(() => ({
+    enter: () => setMouseEnter(true),
+    leave: () => setMouseEnter(false),
+    click: (num: number) => setRating(num),
+    setActive: () => setActive?.(),
+  }), [setActive]);
```

Comment:
setActive 가 뭔가 싶었는데... click 이벤트 trigger 시에 호출해주는 애였네요.
차라리 onClick 으로 바꾸고 아래처럼 하는게 나아보여요

```suggestion
  const value = useMemo(() => ({
    handleMouseEnter: () => setMouseEnter(true),
    handleMouseLeave: () => setMouseEnter(false),
    handleClick: (num: number) => {
      setRating(num);
      onClick?.()
    },
  }), [onClick]);
```

Reply:
StarRating 이 setActive 라는 prop 을 받는게 어색하게 느껴지네요.
onClick 을 받도록 이름을 수정해주세요

Reply:
아 두 함수 모두 Click이벤트 발생시에 일어나는거니 묶어줄 수 있었네요! 감사합니다